reports/gitlab-commit-protocol/gitlab-commit-protocol.tex

```diff
@@ -65,6 +69,18 @@
 \maketitle
 
 \section{Introduction}
+
+Gitlab team pays attention on fact that number of users, repositories, and activity grows, it is important to scale Gitaly appropriately by:
+
+\begin{enumerate}
+\item Increasing the available CPU and memory resources available to Git before resource exhaustion degrades Git, Gitaly, and GitLab application performance.
+\item Increase available storage before storage limits are reached causing write operations to fail.
+\item Improve fault tolerance by removing single points of failure. Git should be considered mission critical if a service degradation would prevent you from deploying changes to production.
+\end{enumerate}
+
+Despite the fact that the gitaly team claims strong consistency, there are possible cases when inconsistency can occur. The main goals of this technical report are to describe full gitlab  workflow of some mutator operation like "git push" and to figure out the distribute fetch process, to find possible cases when inconsistency can be hit. This research work can help us to find the most  relevant protocol to achieve strong consistency in Degitx distributed git repository manager.
```

Comment:
Please try to split long lines to multiple linse in `tex` sources, now it's not possible to comment on specific line.

> there are possible cases when inconsistency can occur

Did we really find any "inconsistency"? 

> "git push"

it should be:
```tex
``git push''
```
in sources: double ` - for opening quote, double ' - for closing.